gen/sshkeys_generic

```diff
@@ -8,44 +8,42 @@ use File::Basename;
 
 local $::SERVICE_NAME = basename($0);
 local $::PROTOCOL_VERSION = "3.0.0";
-my $SCRIPT_VERSION = "3.0.0";
+my $SCRIPT_VERSION = "3.0.1";
 
 perunServicesInit::init;
-my $data = perunServicesInit::getHierarchicalData;
-my $agent = perunServicesInit::getAgent;
+my $data = perunServicesInit::getHashedHierarchicalData;
 my $DIRECTORY = perunServicesInit::getDirectory;
 
 
 our $A_SSHKEYS;                       *A_SSHKEYS =                          \'urn:perun:user:attribute-def:def:sshPublicKey';
-our $A_USER_STATUS;                   *A_USER_STATUS =                      \'urn:perun:member:attribute-def:core:status';
+our $A_MEMBER_STATUS;                 *A_MEMBER_STATUS =                    \'urn:perun:member:attribute-def:core:status';
 our $A_RESOURCE_SSHKEYS_TARGET_USER;  *A_RESOURCE_SSHKEYS_TARGET_USER =     \'urn:perun:resource:attribute-def:def:sshkeysTargetUser';
 
 my $sshkeysDirectory = "$DIRECTORY/$::SERVICE_NAME";
 mkdir $sshkeysDirectory or die "$sshkeysDirectory directory can't be created: $!";
 
-my @resourcesData = $data->getChildElements;
 my $fileStructureWithData;
-foreach my $resourceData (@resourcesData) {
-	my %resourceAttributes = attributesToHash $resourceData->getAttributes;
-	#get information about name of target user (file name for data)
-	my $fileName = $resourceAttributes{$A_RESOURCE_SSHKEYS_TARGET_USER};
-	foreach my $memberData ($resourceData->getChildElements) {
-		my %memberAttributes = attributesToHash $memberData->getAttributes;
+foreach my $resourceId ($data->getResourceIds()) {
+	my $fileName = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_RESOURCE_SSHKEYS_TARGET_USER);
+	foreach my $memberId ($data->getMemberIdsForResource( resource => $resourceId )) {
+		my $userStatus = $data->getMemberAttributeValue(member => $memberId, attrName => $A_MEMBER_STATUS);
```

Comment:
Since you changed ``A_USER_STATUS`` to ``A_MEMBER_STATUS`` I would suggest to change also destination variable from ``userStatus`` to ``memberStatus``.

Reply:
Good point, I didn't notice that variable. Changed.